internal/service/federatedsettingsidentityprovider/model_federated_settings_identity_provider.go

```diff
@@ -22,7 +23,7 @@ func ExpandIdentityProviderOIDCCreate(d *schema.ResourceData) *admin.FederationO
 		Description:       conversion.StringPtr(d.Get("description").(string)),
 		DisplayName:       conversion.StringPtr(d.Get("name").(string)),
 		GroupsClaim:       conversion.StringPtr(d.Get("groups_claim").(string)),
-		IdpType:           conversion.StringPtr(WORKFORCE),
+		IdpType:           conversion.StringPtr(d.Get("idp_type").(string)),
```

Comment:
[q] checking if I understand this right: so if a user has a template without `idp_type` and updates their terraform version, they'll start getting an error as it's no longer defaulted, right? but I guess that's covered by the upgrade-guide?

Reply:
This is only used in the Create, which does not exists in the previous versions of the provider. Once we release this, if a user wants to create a OIDC IdP (WORKFORCE or WORKLOAD) the idp_type is required and will be getting an error if not set in the configuration. But this won't affect users which were using the previous versions of the provider where only Update and Read operations were possible. Let me know if it's clear